docs/next/guides/building-and-testing/plugins.md

```diff
@@ -17,228 +15,242 @@ tags:
+    // The `BasePlugin.disablePlugin` method takes care of clearing the hook connections
+    // and assigning the 'false' value to the 'this.enabled' property.
     super.disablePlugin();
   }
 
   /**
-   * The updatePlugin method is called on the afterUpdateSettings hook (unless the updateSettings method turned the plugin off).
-   * It should contain all the stuff your plugin needs to do to work properly after the Handsontable instance settings were modified.
+   * The updatePlugin method is called on the `afterUpdateSettings` hook
+   * (unless the updateSettings method turned the plugin off).
+   *
+   * It should contain all the stuff your plugin needs to do to work correctly
+   * after updating Handsontable instance settings.
    */
   updatePlugin() {
+    // The `updatePlugin` method needs to contain all the code needed to properly re-enable the plugin.
+    // In most cases simply disabling and enabling the plugin should do the trick.
+    const { enabled, msg } = this.getUnifiedConfig();
 
-    // The updatePlugin method needs to contain all the code needed to properly re-enable the plugin. In most cases simply disabling and enabling the plugin should do the trick.
-    this.disablePlugin();
-    this.enablePlugin();
+    // You can control if settings' update should go through disable->enable routine.
+    if (enabled === false && this.enabled === true) {
+      this.disablePlugin();
+
+    } else if (enabled === true && this.enabled === false) {
+      this.enablePlugin();
+    }
+
+    // Because sometimes the only action you need is to update only one option.
+    if (this.configuration !== null && msg && this.configuration.msg !== msg) {
+      this.configuration.msg = msg;
+    }
 
     super.updatePlugin();
   }
 
+  /**
+   * Define your external methods.
+   */
+  externalMethodExample() {
+    // Method definition.
+  }
+
   /**
    * The afterChange hook callback.
    *
-   * @param {Array} changes Array of changes.
-   * @param {String} source Describes the source of the change.
+   * @param {CellChange[]} changes An array of changes.
+   * @param {string} source Describes the source of the change.
    */
   onAfterChange(changes, source) {
     // afterChange callback goes here.
+    console.log(
+      `${CustomPlugin.PLUGIN_KEY}.onAfterChange - ${this.configuration.msg}`,
+      changes,
+      source
+    );
   }
 
   /**
-   * The destroy method should de-assign all of your properties.
+   * The `destroy` method is the best place to clean up any instances,
+   * objects or index mappers created during the plugin's lifecycle.
    */
   destroy() {
-    // The super method takes care of de-assigning the event callbacks, plugin hooks and clearing all the plugin properties.
+    // The super method takes care of de-assigning the event callbacks,
+    // plugin hooks and clearing all the plugin properties.
     super.destroy();
```

Comment:
// The \`super\` method cleans up the plugin's event callbacks,
// hook connections, and properties.